dom.bs

```diff
@@ -8273,29 +8290,53 @@ steps:
      <li>If <var>range</var>'s
      <a for="live range">root</a> is not equal to
      <var>node</var>'s <a for=tree>root</a>,
-     or if <var>bp</var> is
-     <a for="boundary point">after</a> the
-     <var>range</var>'s <a for=range>end</a>, set
-     <var>range</var>'s <a for=range>end</a>
+     set <var>range</var>'s <a for=range>end</a>
      to <var>bp</var>.
 
+     <li>Otherwise, if <var>bp</var> is
+     <a for="boundary point">after</a> the
+     <var>range</var>'s <a for=range>end</a>,
+     set <var>range</var>'s <a for=range>end</a>
+     to <var>bp</var>. If <var>range</var> is the
+     <a for="composed live range">cached live range</a> of a
+     <a>composed live range</a> <var>composed live range</var>,
+     set <var>composed live range</var>’s
+     <a for="range">end</a> to <var>bp</var>.
+
      <li>Set <var>range</var>'s
      <a for=range>start</a> to <var>bp</var>.
+     If <var>range</var> is the
+     <a for="composed live range">cached live range</a> of a
+     <a>composed live range</a> <var>composed live range</var>,
```

Comment:
We can pull the _composed live range_ variable out early on. How about we detect it and set this variable right after step 2 or step 3, before we branch off into the "set the start" or "set the end" step 4 branches? That should be less redundant.